x-pack/legacy/plugins/siem/server/lib/detection_engine/types.ts

```diff
@@ -49,7 +51,7 @@ export interface RuleAlertParams {
   threat: ThreatParams[] | undefined | null;
   type: 'query' | 'saved_query';
   version: number;
-  throttle?: string;
+  throttle: 'no_actions' | 'rule' | '1h' | '1d' | '7d' | undefined | null;
```

Comment:
I would not invent new branching logic based on strings such as `no_actions` or `rule`. Instead I would pick the value of `null` and use that. I will put comments in other spots for this but we don't want to have branching logic on too many sentinel values and we want to avoid creating new sentinel values when we already have existing paradigms for alerting that are using sentinel values.

Also, you are forcing the throttle to be these "fixed values" and I see the same thing in the schema which is not how we want the REST API be. Rather what you want to do is to just make this:

```ts
throttle: string | null
```

And then default throttle in the schema to be a `null` value and use that as a sentinel value. I will add this advice in the schema below.